ocp_resources/resource.py

```diff
@@ -134,9 +133,27 @@ def sub_resource_level(current_class: Any, owner_class: Any, parent_class: Any)
     return None
 
 
-# Exceptions classes
+def change_dict_value_to_hashed(resource_dict: Dict[Any, Any], key_name: str) -> Dict[Any, Any]:
+    """
+    Recursively search a nested dictionary for a given key and changes its value to "******" if found.
+
+    Args:
+        resource_dict: The nested dictionary to search.
+        key_name: The key to find.
 
-# End Exceptions classes
+    Returns:
+        The modified dictionary.
+    """
```

Comment:
we need to check that arg is dict

```
    if not isinstance(resource_dict, dict):
        raise ValueError("Expected a dictionary as the first argument")
```

Reply:
Ok, will add. But should we talk on this first? I am not sure how to preserve what we have with this PR.

Reply:
This is using recursion. So raising ValueError may not be a good idea.

Reply:
I will do this one. My mistake.

Reply:
Moving this to hash_resource_dict()